skbio/alignment/tests/test_tabular_msa.py

```diff
@@ -493,6 +525,350 @@ def test_reindex_non_hashable_keys(self):
 
         npt.assert_array_equal(msa.keys, keys)
 
+    def test_sort_no_msa_keys_and_key_not_specified(self):
+        msa = TabularMSA([])
+        with self.assertRaises(OperationError):
+            msa.sort()
+        # original state is maintained
+        self.assertEqual(msa, TabularMSA([]))
+
+        msa = TabularMSA([DNA('TC'), DNA('AA')])
+        with self.assertRaises(OperationError):
+            msa.sort()
+        self.assertEqual(msa, TabularMSA([DNA('TC'), DNA('AA')]))
+
+    def test_sort_on_unorderable_msa_keys(self):
+        # TODO is there a better way to handle this?
```

Comment:
Couldn't figure out a better way to get this test to pass on Python 2 and 3. @ebolyen any ideas?